nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FlattenJson.java

```diff
@@ -157,25 +203,36 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final String mode = context.getProperty(FLATTEN_MODE).getValue();
         final FlattenMode flattenMode = getFlattenMode(mode);
 
-        String separator = context.getProperty(SEPARATOR).evaluateAttributeExpressions(flowFile).getValue();
-
+        final Character separator = context.getProperty(SEPARATOR).evaluateAttributeExpressions(flowFile).getValue().charAt(0;
```

Comment:
Thanks for making the updates @naddym, looks like the automated builds failed due to missing a trailing `)` on this line.
```suggestion
        final Character separator = context.getProperty(SEPARATOR).evaluateAttributeExpressions(flowFile).getValue().charAt(0);
```

Reply:
Oh, Somehow missed it while copying after test. Thanks again for commenting.